StringCalculator/StringCalculator.cs

```diff
@@ -0,0 +1,98 @@
+﻿namespace StringCalculator
+{
+    public class StringCalculator
+    {
+        private const int maxNumberToAdd = 1000;
+
+        public int Add(string numbersGiven)
+        {
+            if (string.IsNullOrEmpty(numbersGiven))
+            {
+                return 0;
+            }
+
+            string[] splitNumbersToAdd = SplitStringWithDefaultOrCustomSeperators(numbersGiven);
+
+            ValidateNumbersForNegatives(splitNumbersToAdd);
+
+            string[] splitAndFilteredNumbers = FilterNumbersToSumUnderLimit(splitNumbersToAdd);
+
+            int finalResult = CalculateSplitNumberAddition(splitAndFilteredNumbers);
+
+            return finalResult;
+        }
+
+        private string[] SplitStringWithDefaultOrCustomSeperators(string numbersToSplit)
+        {
+            string numbersToCheckForSeperstors = numbersToSplit;
+
+            string[] seperators = new string[] { ",", "\\n" };
+
+            if (numbersToCheckForSeperstors.StartsWith("//"))
+            {
+                string seperatorFromInput = numbersToSplit[2..numbersToSplit.IndexOf("\\n")];
+                seperators = seperatorFromInput.Split(new string[] { "[", "]" }, StringSplitOptions.RemoveEmptyEntries);
+
+                numbersToCheckForSeperstors = numbersToSplit.Substring(numbersToSplit.IndexOf("\\n") + 2);
+            }
+
+            string[] numbersSplit = numbersToCheckForSeperstors.Split(seperators, StringSplitOptions.None);
+
+            return numbersSplit;
+        }
+
+        private void ValidateNumbersForNegatives(string[] numbersToValidate)
+        {
+            List<string> negativeNumbers = new List<string>();
+
+            foreach (string currentNumber in numbersToValidate)
+            {
+                if (string.IsNullOrEmpty(currentNumber))
+                {
+                    continue;
+                }
+
+                int convertedCurrentNumber = Convert.ToInt32(currentNumber);
+
+                if (convertedCurrentNumber < 0)
+                {
+                    negativeNumbers.Add(currentNumber);                   
+                }
+            }
+
+            if (negativeNumbers.Count > 0)
+            {
+                string message = "Negatives not allowed: " + string.Join(",", negativeNumbers);
+                throw new Exception(message);
+            }
+        }
+
+        public string[] FilterNumbersToSumUnderLimit(string[] numbersToFilter)
+        {
+            string[] numbersFiltered = numbersToFilter.
+                Where(filterCheck => !string.IsNullOrEmpty(filterCheck) && Convert.ToInt32(filterCheck) <= maxNumberToAdd)
```

Comment:
For your attempts going forward, I don't want you touse Linq extension methods. It's not because Linq is bad or because you shouldn't being using Linq, but because it does a lot for you which I want you to think through on your own.

Reply:
Also, it's important to know how the methods work if you're going to be using them :)